### PR TITLE
requirements: update dependencies

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----

### CODE REVIEW

**Comment by mr-cal** [comment]
Had to do the same recently for `codespell`: https://github.com/canonical/craft-providers/commit/ab325f27215ae669e6a9f41c507fa79a28f33b19#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52R7

**Comment by mr-cal** [comment]
Precautiously avoiding the megabump for `setuptools`?

**Comment by tigarmo** [comment]
I don't know about `setuptools`, I'm running `make freeze-requirements` as a separate PR here because I want to add `ruff` and don't want these "other" updates to pollute the ruff PR
what megabump?

**Comment by mr-cal** [comment]
Gotcha.  By megabump, I mean the `49.6.0` to `65.5.1` bump here: https://github.com/snapcore/snapcraft/pull/4007

**Review by mr-cal — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ test-black:
 
 .PHONY: test-codespell
 test-codespell:
-	codespell --quiet-level 4 --ignore-words-list crate,keyserver,comandos,ro --skip '*.tar,*.xz,*.zip,*.bz2,*.7z,*.gz,*.deb,*.rpm,*.snap,*.gpg,*.pyc,*.png,*.ico,*.jar,*.so,changelog,.git,.hg,.mypy_cache,.tox,.venv,venv,_build,buck-out,__pycache__,build,dist,.vscode,parts,stage,prime,test_appstream.py,./snapcraft.spec,./.direnv,./.pytest_cache'
+	codespell --quiet-level 4
```
Comment (lengau) [inline Makefile:18]: If we're moving this configuration to `setup.cfg` it might be better to include the `quiet-level` there too.

**Review by lengau — APPROVED** [review]

File: requirements.txt [hunk 2]
```diff
@@ -1,72 +1,74 @@
...
+more-itertools==9.0.0
 mypy-extensions==0.4.3
-oauthlib==3.2.1
-overrides==7.0.0
-packaging==21.3
-platformdirs==2.5.2
+oauthlib==3.2.2
+overrides==7.3.1
+packaging==23.0
+platformdirs==2.6.2
```
Comment (tigarmo) [inline requirements.txt:35]: heads-up that this update here will change the location of the logs

**Pull request merged**